### PR TITLE
feat: pixi-build backend upgrade

### PR DESCRIPTION
**Pull request by tdejager**
Upgrade the build types.

Still depends on some git sources, but should be fine 😎

### CODE REVIEW

File: crates/pixi-build/src/bin/pixi-build-python/python.rs [hunk 1]
```diff
@@ -267,9 +265,7 @@ impl PythonBuildBackend {
         work_directory: &Path,
     ) -> miette::Result<BuildConfiguration> {
         // Parse the package name from the manifest
-        let Some(name) = self.manifest.workspace.workspace.name.clone() else {
-            miette::bail!("a 'name' field is required in the project manifest");
-        };
+        let name = self.manifest.workspace.workspace.name.clone();
```
Comment (baszalmstra) [inline crates/pixi-build/src/bin/pixi-build-python/python.rs:268]: We shoukd use self.manifest.package for these things!

File: Cargo.toml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jsonrpc-core = "18.0.0"
 
 url = "2.5.3"
 
-rattler-build = { git = "https://github.com/nichmor/rattler-build", branch = "feat/pixi-build", default-features = false }
+rattler-build = { git = "https://github.com/tdejager/rattler-build", branch = "feat/upgrade-middleware", default-features = false }
```
Comment (nichmor) [inline Cargo.toml:34]: should we wait first for the other PR so we can use prefix-dev/rattler-build here?

Comment (tdejager) [reply Cargo.toml:34]: Maybe, lets see if it lands quickly, as we are not publishing this I dont mind it too much. But it would be better for sure.

Comment (tdejager) [reply Cargo.toml:34]: Ok its in lets change to main.

Comment (nichmor) [reply Cargo.toml]: updated to main

**Pull request merged**